test/resource_test.rb

```diff
@@ -133,6 +176,119 @@ def test_acquire_with_fork
     end
   end
 
+  def test_quota_acquire
+    quota = 0.5
+    workers = 9
+
+    resource = create_resource :testing, quota: quota, timeout: 0.1
+    fork_workers(count: workers, quota: quota, wait_for_timeout: true)
+
+    # Ensure that the number of tickets is correct and none are remaining
+    assert_equal quota * (workers + 1), resource.tickets
+    assert_equal 0, resource.count
+
+    # Ensure that no more tickets may be allocated
+    assert_raises Semian::TimeoutError do
+      resource.acquire {}
+    end
+
+    signal_workers('TERM')
+
+    # Ensure the correct number of processes timed out
+    timeouts = Process.waitall.count { |s| s.last.exitstatus == 100 }
+    assert_equal workers - ((1 - quota) * workers).ceil, timeouts
+
+    # Ensure that the tickets were released
+    assert_equal resource.tickets, resource.count
+  end
+
+  def test_quota_increase
```

Comment:
@sirupsen you're quite right.

I think these were really testing two different things here, so I added additional tests and clarified the names:

* test_quota_increase / test_quota_decrease - just spawn a single worker to mutate the quota and make assertions about the ticket count changes. Added the test_quota_increase, to mirror test_quota_decrease.
* test_quota_acquire_increase_quota / test_quota_acquire_decrease_quota - these also take into account how the quota adjustment should affect acquire and timeouts. Renamed test_quota_increase to test_quota_acquire_increase_quota, and wrote test_quota_acquire_decrease_quota to mirror it.